### PR TITLE
core(fr): add `supportedModes` filter to categories

### PR DESCRIPTION
**Pull request by adamraine**
This is a fancy way of hiding PWA in timespan and snapshot. Filter is defined in the config.

### CODE REVIEW

File: flow-report/test/summary/category-test.tsx [hunk 1]
```diff
@@ -8,14 +8,14 @@ import {FunctionComponent} from 'preact';
...
 import {SummaryTooltip} from '../../src/summary/category';
-import {Util} from '../../../report/renderer/util';
 import {flowResult} from '../sample-flow';
 import {I18nProvider} from '../../src/i18n/i18n';
 import {FlowResultContext} from '../../src/util';
 
 let wrapper: FunctionComponent;
 
 beforeEach(() => {
+  // Include sample flowResult for locale in I18nProvider.
```
Comment (adamraine) [inline flow-report/test/summary/category-test.tsx:18]: Drive by change to this test so the expectations don't change every time the sample is updated.

File: lighthouse-core/scripts/update-flow-fixtures.js [hunk 2]
```diff
@@ -13,7 +13,10 @@ import {LH_ROOT} from '../../root.js';
 import UserFlow from '../fraggle-rock/user-flow.js';
 
 (async () => {
-  const browser = await puppeteer.launch({headless: false});
+  const browser = await puppeteer.launch({
+    executablePath: process.env.CHROME_PATH,
```
Comment (adamraine) [inline lighthouse-core/scripts/update-flow-fixtures.js:17]: I was noticing the FPS was rendered incorrectly on the default puppeteer chrome. We could also update the puppeteer version in this PR. Not sure it started showing up now though.

File: types/config.d.ts [hunk 3]
```diff
@@ -134,6 +134,7 @@ declare module Config {
     auditRefs: AuditRefJson[];
     description?: string | IcuMessage;
     manualDescription?: string | IcuMessage;
+    supportedModes?:  Gatherer.GatherMode[];
```
Comment (adamraine) [inline types/config.d.ts:137]: Required in v9?

**Comment by Mbellsudteen** [comment]
ในวันที่ ส. 2 ต.ค. 2021 06:54 น. Adam Raine ***@***.***>
เขียนว่า:

**Comment by Mbellsudteen** [comment]
ในวันที่ ส. 2 ต.ค. 2021 06:53 น. Adam Raine ***@***.***>
เขียนว่า:

**Comment by Mbellsudteen** [comment]
ในวันที่ ส. 2 ต.ค. 2021 06:52 น. Adam Raine ***@***.***>
เขียนว่า:

**Comment by Mbellsudteen** [comment]
ในวันที่ ส. 2 ต.ค. 2021 06:51 น. Adam Raine ***@***.***>
เขียนว่า:

File: types/config.d.ts (hunk 3 above)
Comment (patrickhulce) [reply types/config.d.ts:137]: meh, leaving as the default to always run seems fine even in v9, but I'll leave it up to you

**Review by patrickhulce — APPROVED** [review]

**Comment by patrickhulce** [comment]
sample JSON needs an update

**Comment by brendankenny** [comment]
Thinking out loud but not sure if there's something worthwhile here:

Something about the fact that this is largely about filling in the filtering gaps created because an audit can be in more than one category, and that there's redundancy with `onlyCategories`, makes this feel like a config structural problem that an additional option will add complexity to, rather than simplify.

I understand sharing the default config between the classic and new navigation modes, but is there a reason we haven't made a `default-timespan-config.js` and `default-snapshot-config.js` yet? e.g. rather than implicit filtering, make config entry points, either with `extends: 'lighthouse:default'`/`onlyCategories` or more ambitious reorgs (import what a config needs, `require('./perf-category.js')` or whatever). I don't know if that really fixes the structural issue since it's essentially functionally equivalent, but it would avoid a new config option and it may be more in the spirit of LH supporting config file**s** (plural :).

**Comment by adamraine** [comment]
> I understand sharing the default config between the classic and new navigation modes, but is there a reason we haven't made a default-timespan-config.js and default-snapshot-config.js yet?

This would mean we need 6 configs (desktop/mobile and 3 gather modes for each). We would probably have to repeat some config implementation somewhere.

Maybe we could have used separate configs for (https://github.com/GoogleChrome/lighthouse/pull/13013), but I think the problem is that we don't want simulated throttling to even be possible with timespan modes. I'm not sure we even want PWA to be available at all in snapshot or timespan through `onlyCategories`.

Maybe we could hardcode an override for PWA like we did with throttling settings. With this however, a new audit in SEO that works in all three modes would quietly add SEO to all timespan reports (unless we hardcode another exception).

I like the ability to explicitly control which categories appear for each mode, and I don't think a `supportedModes` filter is any more complicated than having 6 default configs.

**Comment by patrickhulce** [comment]
> is there a reason we haven't made a default-timespan-config.js and default-snapshot-config.js yet? e.g. rather than implicit filtering, make config entry points, either with extends: 'lighthouse:default'/onlyCategories or more ambitious reorgs (import what a config needs, require('./perf-category.js') or whatever).

My primary thoughts here *in general* are that relying on manual configuration rather than the property-based enforcement we see in this PR complicates the story not just for us but all other config authors as well. 

For example, a plugin author would need to publish 3 different versions of their plugin for each gather mode, three npm namespaces, with different configs, etc.

Relying on manual configuration instead of automatic enforcement also tends to be more prone to errors (oops we forgot to update the timespan-config rather than "oh hey we adjusted this audit, adjust the top of the same file")

**Comment by brendankenny** [comment]
Ha, well if y'all feel that good about this, that's fine :)

Long term, I don't think six default configs is that big of a deal. We have three pretty different modes, we have multiple methods of config composition available to us to make maintenance easier (our `extends`, but they're also just JS objects), and presumably the three desktop variants are just going to override the emulation settings of the mobile versions and that's it.

I suspect the more the different modes diverge in expected or required behavior the less desirable it's going to be to smooth over the differences with more configuration options and the easier it's going to be to just define them separately, but obviously this isn't the straw that breaks that particular camel's back :) An option to keep in our back pocket, though.

**Pull request merged**